### PR TITLE
fix: ESLint 9 migration, CI upgrades, and test stability

### PR DESCRIPTION
**Pull request by guyi2000**
## Summary
- Upgrade all GitHub Actions to latest versions (checkout v6, setup-node v6, upload-artifact v7, github-script v9)
- Migrate backend ESLint from `.eslintrc.cjs` to flat config `eslint.config.mjs` (ESLint 9 + typescript-eslint 8)
- Fix flaky `llm-logger-coverage.test.mjs` by replacing fixed `setTimeout` delays with polling `waitForFile`
- Resolve Node 24 DEP0190 warning in `postinstall.js` by invoking prisma directly via `execFileSync`

## Impacted areas
- `backend` (ESLint config, package.json, test)
- `scripts` (postinstall)
- `.github/workflows` (all 6 workflow files)

## Test plan
- [ ] Backend regression CI passes (Ubuntu + Windows)
- [ ] Install smoke CI passes
- [ ] ESLint runs without errors: `npm run lint --prefix backend`
- [ ] No DEP0190 warning on `npm install`

### CODE REVIEW

File: backend/eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import js from "@eslint/js";
+import tseslint from "typescript-eslint";
+
+export default [
+  // 1️⃣ 基础 JS 推荐规则
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/eslint.config.mjs:9]: Fixed in eab0c7b — added `ignores: ["dist/", "node_modules/"]` at the top of the flat config array.

File: backend/eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      parserOptions: {
+        ecmaVersion: "latest",
+        sourceType: "module",
+      },
+    },
+
+    rules: {
+      // 常见自定义
+      "no-unused-vars": "off",
+      "@typescript-eslint/no-unused-vars": ["warn"],
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/eslint.config.mjs]: Fixed in eab0c7b — restored `error` severity and `argsIgnorePattern: "^_"` to match the original `.eslintrc.cjs`.

File: backend/tests/llm-logger-coverage.test.mjs [hunk 3]
```diff
@@ -27,6 +27,26 @@ function readJsonlLines(filePath) {
...
+        }
+      } catch { /* ignore */ }
+      if (Date.now() - start >= timeoutMs) {
+        reject(new Error(`Timed out waiting for ${filePath}`));
+        return;
+      }
+      setTimeout(check, intervalMs);
+    };
+    check();
+  });
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/tests/llm-logger-coverage.test.mjs]: Fixed in eab0c7b — `waitForFile` now accepts an `expectedLines` option and polls until the parsed line count reaches the target before resolving. Multi-entry tests pass `expectedLines: 3` and `expectedLines: 50` respectively.

File: backend/tests/llm-logger-coverage.test.mjs [hunk 4]
```diff
@@ -539,18 +557,10 @@ describe('llmCallLogger.log edge cases', () => {
...
-          lines = readJsonlLines(jsonlPath);
-        } catch {
-          lines = [];
-        }
-        if (lines.length >= 50) break;
-      }
+      await waitForFile(jsonlPath);
+      const lines = readJsonlLines(jsonlPath);
       expect(lines).toHaveLength(50);
       expect(lines[0].model).toBe('model-0');
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/tests/llm-logger-coverage.test.mjs:572]: Fixed in eab0c7b — the 50-entry test now uses `waitForFile(jsonlPath, { expectedLines: 50 })`.

File: backend/eslint.config.mjs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+      "@typescript-eslint/consistent-type-imports": "warn",
+      "@typescript-eslint/no-explicit-any": "off",
+    },
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/eslint.config.mjs:32]: Fixed in eab0c7b — restored `error` severity and `argsIgnorePattern: "^_"` to match the original config. Also fixed the only catch-clause violation by removing the unused parameter.

**Review by qinsz01 — APPROVED** [review]
